presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java

```diff
@@ -566,6 +573,53 @@ static String formatSortItems(List<SortItem> sortItems, boolean unmangleNames)
                 .iterator());
     }
 
+    static String formatGroupBy(List<GroupingElement> groupingElements)
+    {
+        ImmutableList.Builder<String> resultStrings = ImmutableList.builder();
+
+        for (GroupingElement groupingElement : groupingElements) {
```

Comment:
I think this can all be simplified a bit if you use formatGroupingSet in every place where you're printing a list of columns (and update that method to handle the single-column case). After all, a grouping set of a single element can be formatted without parenthesis. This applies for the simple group by case, as well as the other cases:

`CUBE a`
`ROLLUP a`
`GROUPING SETS (a)`